staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache_test.go

```diff
@@ -486,6 +516,21 @@ func TestWaitUntilFreshAndList(t *testing.T) {
 	if indexUsed != "" {
 		t.Errorf("Used index %q but expected none to be used", indexUsed)
 	}
+	// list from future revision. Requires watch cache to request bookmark to get it.
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ConsistentListFromCache, true)()
```

Comment:
Instead of growing existing already long test, can we make a separate test from it?

Reply:
Done